sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/CloudEventRequestContent.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Azure.Core.Serialization;
+
+namespace Azure.Messaging.EventGrid
+{
+    internal class CloudEventRequestContent : RequestContent
+    {
+        private IEnumerable<CloudEvent> _cloudEvents;
+        private static readonly JsonObjectSerializer s_jsonSerializer = new JsonObjectSerializer();
```

Comment:
Why don't we use the JsonSerializer directly?

Reply:
We can.. is there any benefit?

Reply:
Less overhead.

Reply:
Well we would still have to serialize to a stream, unless we use the byte overload that doesn't support async.

Reply:
Nevermind I guess there is no reason to use the async here.